configs/components/puppet.rb

```diff
@@ -235,4 +236,55 @@
   if platform.is_eos?
     pkg.link "#{settings[:sysconfdir]}", "#{settings[:link_sysconfdir]}"
   end
+
+  pkg.install_file "ext/hiera/hiera.yaml", File.join(settings[:puppet_codedir], 'environments', 'production', 'hiera.yaml')
+  pkg.configfile File.join(settings[:puppet_codedir], 'environments', 'production', 'hiera.yaml')
+  pkg.configfile File.join(configdir, 'hiera.yaml')
+
+  unless platform.is_windows?
+    old_hiera = File.join(settings[:puppet_codedir], 'hiera.yaml')
+    cnf_hiera = File.join(configdir, 'hiera.yaml')
+    env_hiera = File.join(settings[:puppet_codedir], 'environments', 'production', 'hiera.yaml')
+    rmv_hiera = File.join(configdir, 'remove_hiera5_files.rm')
+
+    # Pre-Install script to save copies of existing hiera configuration files.
+    # This also creates "marker files to indicate that the files pre-existed."
+    preinstall = <<-PREINST
+# Backup the old hiera configs if present, so that we
+# can drop them back in place if the package manager
+# tries to remove it.
+if [ -e #{old_hiera} ]; then
+  cp #{old_hiera}{,.pkg-old}
+  touch #{rmv_hiera}
+fi
+if [ -e #{cnf_hiera} ]; then
+  cp #{cnf_hiera}{,.pkg-old}
+  touch #{rmv_hiera}
+fi
+PREINST
+
+    # Post-install script to restore old hiera config files if the have been saved.
+    # and remove any extre hiera configuration files that we laid down
+    postinstall = <<-POSTINST
+# Remove any extra hiera config files we laid down if prev config present
+if [ -e #{rmv_hiera} ]; then
+  rm -f #{cnf_hiera}
+  rm -f #{env_hiera}
+  rm -f #{rmv_hiera}
+fi
```

Comment:
With the logic in this script, if the file `/etc/puppetlabs/code/hiera.yaml` exists, but the file `/etc/puppetlabs/puppet/hiera.yaml` does not, we will:
- back up, overwrite, and remove, and restore `/etc/puppetlabs/code/hiera.yaml`
- write and then remove `/etc/puppetlabs/hiera.yaml` 

Leaving the host in a state without any `/etc/puppetlabs/hiera.yaml` (it had none to begin with). I was under the impression we wanted to install the new hiera 5 format yaml to `/etc/puppetlabs/puppet/hiera.yaml` if it does not exist. Is that right?